syntax = "proto3";

package service.proto;

service TaskListService {
  // Создать новую задачу
  rpc CreateTask (CreateTaskRequest) returns (Task);

  // Редактировать существующую задачу
  rpc UpdateTask (UpdateTaskRequest) returns (Task);

  // Просмотреть список задач
  rpc ListTasks (ListTasksRequest) returns (TaskListResponse);

  // Удалить задачу
  rpc DeleteTask (DeleteTaskRequest) returns (Task);
}

message CreateTaskRequest {
  string title = 1;
  string description = 2;
}

message UpdateTaskRequest {
  int32 task_id = 1;
  string title = 2;
  string description = 3;
}

message ListTasksRequest {
  // Здесь можно добавить параметры для фильтрации, если необходимо
}

message Task {
  int32 id = 1;
  string title = 2;
  string description = 3;
}

message TaskListResponse {
  repeated Task tasks = 1;
}

message DeleteTaskRequest {
  int32 task_id = 1;
}
